### DISCUSSION TITLE
Read/Write Remote Emulated PLC?

### DISCUSSION THREAD

**Discussion by Xiadais** [ottowayi/pycomm3#255] [Q&A]
Is it possible to communicate with an emulated PLC on a different computer? I know it's possible to talk to a local emulator with '127.0.0.1/{slot}', and it's theoretically possible to communicate with a remote emulator since RSLinx/OPC/etc can do so.

I'm wondering if this is a simple case which could be fixed with fairly minor additions to recognize the identity response, or if the entire method of communication is different.

Let me know if there's any additional information I could provide to help here.

pycomm3 version 1.2.10
Emulate 5570 controller (v30, not that it matters)

(The .log was made with verbose logging; the .txt is just a Python script with a modified extension to get around GitHub's allowable attachments.)
[pycomm3.log](https://github.com/ottowayi/pycomm3/files/10261387/pycomm3.log)
[testpycomm.txt](https://github.com/ottowayi/pycomm3/files/10261389/testpycomm.txt)

**Comment by ottowayi** [comment]
I think this only works on the latest versions on emulate, it works on my machine with v32:
![image](https://user-images.githubusercontent.com/5274950/208530106-af267372-8945-4240-b275-9dc26343b7fb.png)

Discover shows the IP of the machine and both that and localhost work for connecting.  I think it was only recently they added that and prior to it there was some secret Rockwell way that only their software knew.

**Comment by Xiadais** [comment]
Interesting... CIPDriver.discover() is giving me an empty list, on both the emulating computer and the remote computer. I am 100% sure I have communicated with emulators on different computers, not sure what I didn't set up right...

**Reply by bachphi** [reply]
My VM is running Emulate v35 and my host machine is running Pycomm3 1.2.14. The host can read tags from the Emulate 5000 VM without any issues:
```
from pycomm3 import LogixDriver
with LogixDriver('10.10.10.104/2') as plc:
    print(plc.info)
```
However, pycomm3.CIPDriver.discover() cannot detect the PLC from the host, return empty bracket. So discovery fails, but explicit connections to the VM work normally.
